lib/DB/Schema/ResultSet/Course.pm

```diff
@@ -451,44 +443,30 @@ A single course setting as either a hashref or a C<DBIx::Class::ResultSet::Cours
 =cut
 
 sub updateCourseSetting ($self, %args) {
-	my $course = $self->getCourse(info => getCourseInfo($args{info}), as_result_set => 1);
-
+	my $course         = $self->getCourse(info => getCourseInfo($args{info}), as_result_set => 1);
 	my $global_setting = $self->getGlobalSetting(info => getSettingInfo($args{info}));
 
 	my $course_setting = $course->course_settings->find({
 		setting_id => $global_setting->{setting_id}
 	});
 
-	# Check that the setting is valid.
-
 	my $params = {
 		course_id  => $course->course_id,
 		setting_id => $global_setting->{setting_id},
-		value      => { value => $args{params}{value} }
+		value      => $args{params}{value}
 	};
 
-	# remove the following fields before checking for valid settings:
-	delete $global_setting->{$_} for (qw/setting_id course_id/);
+	isValidSetting($global_setting, $params->{value});
 
-	isValidSetting($global_setting, $params->{value}{value});
-
-	# The course_id must be deleted to ensure it is written to the database correctly.
-	delete $params->{course_id} if defined($params->{course_id});
-
-	my $updated_course_setting =
+	my $up_setting =
 		defined($course_setting) ? $course_setting->update($params) : $course->add_to_course_settings($params);
 
-	return $updated_course_setting if $args{as_result_set};
+	return $up_setting if $args{as_result_set};
 	my $setting_to_return =
 		($args{merged})
-		? { $updated_course_setting->get_inflated_columns,
-			$updated_course_setting->global_setting->get_inflated_columns }
-		: { $updated_course_setting->get_inflated_columns };
+		? { $up_setting->get_inflated_columns, $up_setting->global_setting->get_inflated_columns }
+		: { $up_setting->get_inflated_columns };
```

Comment:
here we can also just return instead of creating `$setting to return`

Reply:
fixed.